book_source/03_topical_pages/07_remote_access/01_pecan_api.Rmd

```diff
@@ -541,6 +553,220 @@ print(json.dumps(response.json(), indent=2))
+##       "format_id": 19,
+##       "format_name": "AmeriFlux.level4.h"
+##     },
+##     {
+##       "mimetype": "text/csv",
+##       "format_id": 20,
+##       "format_name": "AmeriFlux.level4.d"
+##     },
+##     ...
+##   ],
+##   "count": 6
+## }
+  
+```
+
+### {-}
+
+### `GET /api/formats/{format_id}` {.tabset .tabset-pills}
+
+#### R Snippet
+
+```R
+# Fetch the details of PEcAn format with id = 19
+res <- httr::GET(
+         "http://localhost:8000/api/formats/19",
+         httr::authenticate("carya", "illinois")
+       )
+print(jsonlite::fromJSON(rawToChar(res$content)))
+```
+```
+## $mimetype
+## [1] "text/csv"
+
+## $format_id
+## [1] 19
+
+## $name
+## [1] "AmeriFlux.level4.h"
+
+## $notes
+## [1] "Half-hourly AmeriFlux level 4 gap filled, partitioned, and flagged flux tower data. Variables description:  Level 4 data are obtained from the level 3 products, data are ustar filtered, gap-filled using different methods (ANN and MDS) and partitioned (i.e. NEE, GPP, and Re). Flags with information regarding quality of the original and gapfilled data are added.  Missing values: -9999."
+
+## $header
+## [1] ""
+
+## $format_variables
+##                                                   description        name             unit
+## 1                                            Latent heat flux        LE_f            W m-2
+## 2                                          Sensible heat flux         H_f            W m-2
+## 3                                             air temperature        Ta_f        degrees C
+## 4                                      Vapor Pressure Deficit       VPD_f               Pa
+## 5 Cumulative ecosystem respiration over a specified time step        Reco umol C02 m-2 s-1
+## 6                                      Net ecosystem exchange NEE_st_fMDS   umol C m-2 s-1
+```
+
+#### Python Snippet
+
+```python
+# Fetch the details of PEcAn format with id = 19
+response = requests.get(
+             "http://localhost:8000/api/formats/19",
+             auth=HTTPBasicAuth('carya', 'illinois')
+           )
+print(json.dumps(response.json(), indent=2))
+```
+```
+## {
+##   "mimetype": "text/csv",
+##   "format_id": 19,
+##   "name": "AmeriFlux.level4.h",
+##   "notes": "Half-hourly AmeriFlux level 4 gap filled, partitioned, and flagged flux tower data. Variables description:  ## Level 4 data are obtained from the level 3 products, data are ustar filtered, gap-filled using different methods (ANN and ## MDS) and partitioned (i.e. NEE, GPP, and Re). Flags with information regarding quality of the original and gapfilled data ## are added.  Missing values: -9999.",
+##   "header": "",
+##   "format_variables": [
+##     {
+##       "description": "Latent heat flux",
+##       "name": "LE_f",
+##       "unit": "W m-2"
+##     },
+##     ...
+##   ]
+## }
+```
+
+### {-}
+
+### `GET /api/inputs/` {.tabset .tabset-pills}
+
+#### R Snippet
+
+```R
+# Get the inputs needed for a workflow with model_id = 1000000022 & site_id = 676
```

Comment:
as noted above, could we have an example that doesn't have a model_id but does have a format_id?

Reply:
Ok...I will add that